src/gnn_reco/data/utils.py

```diff
@@ -9,6 +9,46 @@
 from typing import List, Tuple
 import sqlite3
 
+
+def GetDesiredEventNumbers(db_path, desired_size, fraction_noise = 0, fraction_nu_e = 0, fraction_muon = 0, fraction_nu_mu = 0, fraction_nu_tau = 0):
+    assert fraction_nu_e + fraction_muon + fraction_nu_mu + fraction_nu_tau + fraction_noise == 1.0, 'Sum of fractions not equal to one.'
+
+    fracs = [fraction_noise, fraction_nu_e, fraction_muon, fraction_nu_mu, fraction_nu_tau]
+    numbers_desired = [int(x * desired_size) for x in fracs]
+    pids = [1,12,13,14,16]
+
+    with sqlite3.connect(db_path) as con:
+        total_query = 'SELECT event_no FROM truth WHERE abs(pid) IN {}'.format(tuple(pids))
+        tot_event_nos = pd.read_sql(total_query,con)
+        if len(tot_event_nos) < desired_size:
+            desired_size = len(tot_event_nos)
+            numbers_desired = [x * desired_size for x in fracs]
+            print('Only {} events in database, using this number instead.'.format(len(tot_event_nos)))
+
+        list_of_dataframes = []
+        restart_trigger = True
+        while restart_trigger:
+            restart_trigger = False
+            for number,particle_type in zip(numbers_desired,pids):
+                query_is = 'SELECT event_no FROM truth WHERE abs(pid) == {} LIMIT {}'.format(particle_type,number)
```

Comment:
By using `LIMIT` here, we are selecting a non-random subset of events. It would probably be better practices (although slower to run) to load in all event numbers for a specific PDG ID and then take a random subset of these.